pauthabielf64/pauthabielf64.rst

```diff
@@ -875,56 +877,96 @@ The following values of the platform id are reserved:
+  by the platform vendor identified by the ``platform identifier``
+  above.
 
-If a file contains a section named ``.note.AARCH64-PAUTH-ABI-tag``,
-it must observe the entirety of the rules in this default marking
-schema. Generating such section with a platform-specific schema is
-forbidden.
+The tuple of (``platform identifier``, ``version number``) equal to
+(0,0) is reserved to mean an ELF file incompatible with the PAuth ABI
+Extension to ELF for the Arm® 64-bit Architecture (AArch64).
+
+Default Marking Schema
+----------------------
+
+The default ELF marking scheme for executables and shared-libraries
+uses the ``.note.gnu.property`` section. The format of this section is
+defined in (`LINUX_ABI`_).
+
+The following processor-specific program property types are defined:
+
+    +----------------------------------------+------------+
+    | Name                                   | Value      |
+    +========================================+============+
+    | GNU\_PROPERTY\_AARCH64\_FEATURE\_PAUTH | 0xc0000001 |
+    +----------------------------------------+------------+
+
+Other processor-specific program property types defined by the 64-bit
+ABI for the Arm Architecture are defined in (SYSVABI64_).
+
+The format of the data in ``pr_data`` is two 64-bit words. With the
+first 64-bit word being the ``platform identifier``, and the second
+64-bit word being the ``version number``. Both of these form the
+information required in `Core Information`_ above.
+
+``.note.gnu.property`` sections can be used as ELF marking for
+relocatable objects as well as executables and shared libraries. Arm
+intends to use standardize build attributes for all relocatable-object
+ELF marking. When this change occurs the default ELF marking for
+relocatable objects will be updated to use build attributes.
 
 Base Compatibility Model
 ------------------------
 
 A per-ELF file marking scheme permits a coarse way of reasoning about
 compatibility.
 
-* The absence of a ``.note.AARCH64-PAUTH-ABI-tag`` section means no
-  information on how pointers are signed is available for this ELF
-  file.
+* All reasoning about compatibility is done using the `Core Information`_.
+  This permits an ELF file using the ``.note.gnu.property`` ELF marking to
+  be compared to an ELF file using the ``.note.AARCH64-PAUTH-ABI-tag`` ELF
+  marking.
+
+* If an ELF file contains multiple ELF markings of the `Core
+  Information`_, for example it contains both a ``.note.gnu.property``
+  section and a ``.note.AARCH64-PAUTH-ABI-tag`` section, then all
+  must encode the same `Core Information`_.
 
-* The presence of a ``.note.AARCH64-PAUTH-ABI-tag`` means that, if the
-  file contains pointers, they were signed in a compatible way with
-  the default signing rules for tuple (platform id, version number).
+* The absence of any ELF marking means no information on how pointers
+  are signed is available for this ELF file. When used in combination
```

Comment:
> When used in combination
> with ELF files that contain ELF marking, then the file is assigned
> the (``platform identifer``, ``version number``) of (0,0).

I think we should allow different behavior here. A user might want static linker to "ignore" such files with no compatibility info and treat them as compatible with any others. See similar option `-z bti-report` in lld https://github.com/llvm/llvm-project/blob/546f32df26f58fdfe02d99e6d91d681dd9ed6839/lld/docs/ld.lld.1#L749. It might be helpful if we already have some pre-built pauth-disabled binaries and if we are sure that using them won't break pauth-enabled code from other binaries. For example, in a C/C++ project, we might have an assembly file manually defining a section or smth, and it would be nice not to include pauth compatibility info inside it manually as well.

Reply:
I've added a clarifying sentence that states that implementations can be provided with supplemental information that can change this default.